front/app/containers/Admin/reporting/components/Tabs.tsx

```diff
@@ -85,8 +85,22 @@ const DashboardTabs = ({ reportBuilderEnabled, children }: Props) => {
 };
 
 const DashboardTabsWrapper = ({ children }: { children: React.ReactNode }) => {
-  const reportBuilderEnabled = useFeatureFlag({ name: 'report_builder' });
-  if (reportBuilderEnabled === undefined) return null;
+  const appConfig = useAppConfiguration();
+
+  if (isNilOrError(appConfig)) {
+    return null;
+  }
+
+  const isReportBuilderAllowed =
+    appConfig.attributes.settings.report_builder?.allowed;
+
+  const isReportBuilderEnabled =
+    appConfig.attributes.settings.report_builder?.enabled;
+
+  // We don't show the report builder tabs if the feature flag is allowed but was intentionally disabled
+  const reportBuilderEnabled = !(
+    isReportBuilderAllowed && !isReportBuilderEnabled
+  );
```

Comment:
This is the same as the code that was there. 

`const reportBuilderEnabled = useFeatureFlag({ name: 'report_builder' });
`

Reply:
Not really, the intention here is that we want to hide the tabs for users who have it allowed but explicitly decided to have `enabled` false. The code that was here was considering both

Reply:
> Not really, the intention here is that we want to hide the tabs for users who have it allowed but explicitly decided to have enabled false. 

Yes, that's exactly what you can check with:

`const reportBuilderEnabled = useFeatureFlag({ name: 'report_builder' }); `

If this is true, it means allowed & enabled are true. If this is false, allowed is true, but enabled is false. If this returns true, allowed should always be true. Only enabled changes.

What you currently check with `reportBuilderEnabled` is the case where `useFeatureFlag({ name: 'report_builder' }) `would return `false`. So either `reportBuilderEnabled` is a wrong variable name or you're checking the wrong thing. I think it's the latter.

I actually think a clearer variable name would be `showReportBuilderTab` so that is decoupled from the condition.

Reply:
Changing the name should be fine but sure, create one and add me as a reviewer